code/implementation/docker/compute/src/main/java/io/cattle/platform/docker/process/instancehostmap/DockerPostInstanceHostMapActivate.java

```diff
@@ -99,21 +108,31 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
 
         processVolumes(instance, host, state);
 
+        nativeDockerBackPopulate(instance);
+
         return null;
     }
 
-    @SuppressWarnings({ "unchecked", "rawtypes" })
+    @SuppressWarnings({ "unchecked" })
+    protected void nativeDockerBackPopulate(Instance instance) {
+        Map<String, Object> inspect = (Map<String, Object>)instance.getData().get(FIELD_DOCKER_INSPECT);
+        if (inspect == null || instance.getNativeContainer() == null || !instance.getNativeContainer().booleanValue()) {
+            return;
+        }
+        transformer.transform(inspect, instance);
```

Comment:
Craig, not sure if it should be a point of concern, but I see that we update instance and then update the same instance with the volumes as a part of 2 separate calls. Would it cause any problem if one of the transactions fail? And if yes, we should probably wrap up these 2 updates within a single transaction